code/drasil-lang/lib/Language/Drasil/Expr/Lang.hs

```diff
@@ -160,6 +160,38 @@ data Expr where
   Operator :: AssocArithOper -> DiscreteDomainDesc Expr Expr -> Expr -> Expr
   -- | A different kind of 'IsIn'. A 'UID' is an element of an interval.
   RealI    :: UID -> RealInterval Expr Expr -> Expr
+  -- | A clif is an expression
+  Clif     :: Clif -> Expr
+
+-- | The basis in which to project clifs
+data Basis where
+  -- | ℝⁿ
+  Rn :: Int -> Basis
+
+-- | A clif is a clif, or the addition of clifs, or the scalar multiplication or geometic product of clifs
```

Comment:
I think it might make sense to separate base `Clif` from `Clif`-forming operations.

Also, in the AST, I don't think it makes a lot of sense to have all those special cases - go straight for `NGradeC`. We can have the external API provide all these special cases (both for input and output).